src/utils/inline-svg.js

```diff
@@ -50,3 +51,20 @@ function makeDom(svgText) {
   wrapper.innerHTML = svgText.trim();
   return wrapper.firstChild;
 }
+
+export async function svgExists(path) {
+  if (path in loaded) {
+    return true;
+  }
+  if (path in doesntExist) {
+    return false;
+  }
+  //not cached yet
+  const response = await fetch(path);
+  const svgText = await response.text();
+  const exists = svgText.startsWith('<svg');
+  if (!exists) {
+    doesntExist[path] = true;
+  }
```

Comment:
Any reason to not also cache it in `loaded`?

Reply:
Since assignHeadAndTailUrls() gets called every frame, svgExists() will also be called every frame, which is why it does caching to avoid running a fetch per head/tail.

If the broken/ non existent svg paths were stored in loaded, then requireSvg(), fetchSvg() and getSvg() would still be retrieving and passing back <html> instead of <svg>'s, and extra checks to catch broken paths would have to be run either:
-  inside each of requireSvg(), fetchSvg(), getSvg()
- or inside setHeadAndTailSvgs() on every frame update

This fix seemed the most optimal way because:  all the DEFAULT_SNAKE_TAIL/DEFAULT_SNAKE_HEAD logic is still handled inside engine-client.js (didn't want to have to add a second place where this default head/tail is enforced, and the existence of the SVGs only has to be checked via a fetch on the first frame to be loaded.

Reply:
👍